src/spring/azext_spring/buildpack_binding.py

```diff
@@ -107,10 +121,8 @@ def _create_app_insights_and_get_connection_string(cmd, resource_group, service_
         created_app_insights = _try_create_application_insights(cmd, resource_group, service_name, location)
         if created_app_insights:
             return created_app_insights.connection_string
-    except Exception:  # pylint: disable=broad-except
-        logger.warning(
-            'Error while trying to create and configure an Application Insights for the Azure Spring Apps. '
-            'Please use the Azure Portal to create and configure the Application Insights, if needed.')
+    except Exception as ex:  # pylint: disable=broad-except
```

Comment:
How about making the logic like this? Because the creation AI is just a fallback solution for fallback solution to get the connection string. It is wired the properties contruct method to handle the creation failure error.

```
def get_properties(sampling_rate, **kwargs):
   conn = get_connection_string(**kwargs)
   if not conn:
        logger.warning('No connection string blablabla')
        return None
   return models.BuildpackBindingProperties(.....)

def get_connection_string(app_insights_key, **kwargs):
    return app_insights_key or get_ai_connection_string(**kwargs) or create_ai_and_get_connection_string(**kwargs)

def create_ai_and_get_connection_string(**kwargs):
    try:
         return create_ai(**kwargs).connection_string
   except Exception as ex:
        logger.warning('failed to create ai blablabla')
        return None
```

Reply:
Yeah, it makes it clearer, updated.